diff --git a/build/rules/global b/build/rules/global
index ed178814d..d9ebb0ca4 100644
--- a/build/rules/global
+++ b/build/rules/global
@@ -84,17 +84,7 @@ endif
 #
 # Determine OS
 #
-POCO_HOST_OSNAME = $(shell uname)
-ifeq ($(findstring CYGWIN,$(POCO_HOST_OSNAME)),CYGWIN)
-POCO_HOST_OSNAME = CYGWIN
-endif
-
-ifeq ($(findstring MINGW,$(POCO_HOST_OSNAME)),MINGW)
-POCO_HOST_OSNAME = MinGW
-endif
-ifdef POCO_VERBOSE
-$(info POCO_HOST_OSNAME    = $(POCO_HOST_OSNAME))
-endif
+POCO_HOST_OSNAME = Linux
 
 POCO_HOST_OSARCH ?= $(subst /,-,$(shell uname -m | tr ' ' _))
 ifdef POCO_VERBOSE
